src/lib/services/events.js

```diff
@@ -0,0 +1,99 @@
+// const moment = require('moment');
+const repo = require('../connectors/repos');
+const { mapValues, mapKeys } = require('lodash');
+const camelCase = require('camelcase');
+const snakeCase = require('snake-case');
+const Event = require('../models/event');
+
+/**
+ * Checks whether the supplied value/key is a JSON field and is not null
+ * @param {Mixed} value
+ * @param {String} key
+ * @return {Boolean} true if key is a JSON field and value is not null
+ */
+const isMappedField = (value, key) => {
+  const jsonFields = ['licences', 'entities', 'metadata'];
+  return jsonFields.includes(key) && (value !== null);
+};
+
+/**
+ * Maps keys of the data using the supplied mapper function
+ * @param  {Object} data     - data to map
+ * @param  {Function} mapper - function to use for mapping keys
+ * @return {Object}        data with keys mapped
+ */
+const keyMapper = (data, mapper) => {
+  return mapKeys(data, (value, key) => mapper(key));
+};
+
+/**
+ * Maps data to the format expected by the repository class
+ * @param  {Object} data - event data
+ * @return {Object}      - event data mapped to DB column names/types
+ */
+const mapToRepo = data => {
+  const mapped = mapValues(data, (value, key) => {
+    return isMappedField(value, key) ? JSON.stringify(value) : value;
+  });
+  return keyMapper(mapped, snakeCase);
+};
+
+/**
+ * Maps keys for data coming from repository class and
+ * returns a new Event model POJO created from the mapped data
+ * @param  {Object} data - data from repository class
+ * @return {Object}      - data with keys camel cased
+ */
+const mapFromRepo = (data) => {
+  const result = keyMapper(data, camelCase);
```

Comment:
If using Bookshelf, the keys should hopefully already be camel-cased

Reply:
done